test-framework/kubernetes-client/src/main/java/io/quarkus/test/kubernetes/client/PortForwardingTestResource.java

```diff
@@ -0,0 +1,128 @@
+package io.quarkus.test.kubernetes.client;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.LocalPortForward;
+import io.quarkus.test.common.QuarkusTestResourceConfigurableLifecycleManager;
+import io.quarkus.test.kubernetes.client.WithPortForwarding.FieldSelector;
+import io.quarkus.test.kubernetes.client.WithPortForwarding.LabelValue;
+
+public class PortForwardingTestResource extends AbstractNamespaceConnectingTestResource
+        implements QuarkusTestResourceConfigurableLifecycleManager<WithPortForwarding> {
+
+    private static final Logger log = LoggerFactory.getLogger(PortForwardingTestResource.class);
+    private LocalPortForward portForward;
+    private String podName;
+    private int port;
+    private int localPort;
+
+    @Override
+    public void init(WithPortForwarding annotation) {
+        initNamespaceAndClient(annotation.namespace());
+
+        final var podIdentifier = annotation.pod();
+
+        // identify which pod to target
+        Pod pod;
+        final var hasLabelSelector = !AnnotationConstants.UNSET_STRING_VALUE.equals(podIdentifier.labelSelector());
+        final var hasFieldSelectors = podIdentifier.fieldSelectors().length > 0;
+        final var hasLabelValues = podIdentifier.labelValues().length > 0;
+        if (!hasLabelSelector && !hasFieldSelectors && !hasLabelValues) {
+            throw new IllegalArgumentException(
+                    "Must specify label values and/or label and/or field selectors to identify which pod to port forward");
+        }
+
+        // retrieve pod name from selectors and optional index
```

Comment:
I think that you would possibly save yourself a lot of trouble if you targeted the `Service` rather than the pod.
Using the pod has two issues:

- You need to know the pod name and you usually don't.
- You need to implement complex code, that you don't really need. Just use the `service` in order to port forward, it's a win win.